src/lib/pickles/opt_sponge.mli

```diff
@@ -19,10 +19,46 @@ module Make
 
   val create : ?init:Impl.Field.t array -> Impl.Field.t Sponge.Params.t -> t
 
+  (** Create a new sponge with state copied from the given sponge.
+      In particular, this copies the underlying state array, so that any
+      mutations to the copy will not affect the original.
+  *)
+  val copy : t -> t
+
   val of_sponge : Impl.Field.t Sponge.t -> t
 
   val absorb :
     t -> Impl.Field.t Snarky_backendless.Boolean.t * Impl.Field.t -> unit
 
   val squeeze : t -> Impl.Field.t
+
+  (** Updates the sponge state by forcing absorption of all 'pending' field
+      elements passed to [absorb].
+      This method runs logic equivalent to that in the [squeeze] method, but
+      without transitioning the state.
+      This method can be used with [copy] to create a fork of a sponge where
+      one of the branches calls the [absorb] method and the other does not.
+  *)
+  val consume_all_pending : t -> unit
+
+  (** Recombines a forked copy of a sponge with the original.
+      When the boolean value is true, the sponge state will be preserved;
```

Comment:
Nice doc!

Nit: It is fine in this case as we can infer the boolean parameter, but when writing documentation, I find more readable to name the parameters and say `When [b] is [true], [...]`. I think starting the doc with:
````
[recombine ~original_sponge preserve_state sponge] recombines a forked copy of a sponge with the original.
When [preserve_state] is [true], ...
```